vars/buildPlugin.groovy

```diff
@@ -41,132 +41,140 @@ def call(Map params = [:]) {
+                                error 'There were test failures; halting early'
                             }
-                            checkstyle(cp)
-                        }
-                        if (failFast && currentBuild.result == 'UNSTABLE') {
-                            error 'There were test failures; halting early'
-                        }
-                        if (doArchiveArtifacts) {
-                            if (incrementals) {
-                                String changelist = readFile(changelistF)
-                                dir(m2repo) {
-                                    fingerprint '**/*-rc*.*/*-rc*.*' // includes any incrementals consumed
-                                    archiveArtifacts artifacts: "**/*$changelist/*$changelist*",
-                                            excludes: '**/*.lastUpdated',
-                                            allowEmptyArchive: true // in case we forgot to reincrementalify
-                                }
-                                publishingIncrementals = true
-                            } else {
-                                String artifacts
-                                if (isMaven) {
-                                    artifacts = '**/target/*.hpi,**/target/*.jpi'
+                            if (doArchiveArtifacts) {
+                                if (incrementals) {
+                                    String changelist = readFile(changelistF)
+                                    dir(m2repo) {
+                                        fingerprint '**/*-rc*.*/*-rc*.*' // includes any incrementals consumed
+                                        archiveArtifacts artifacts: "**/*$changelist/*$changelist*",
+                                                excludes: '**/*.lastUpdated',
+                                                allowEmptyArchive: true // in case we forgot to reincrementalify
+                                    }
+                                    publishingIncrementals = true
                                 } else {
-                                    artifacts = '**/build/libs/*.hpi,**/build/libs/*.jpi'
+                                    String artifacts
+                                    if (isMaven) {
+                                        artifacts = '**/target/*.hpi,**/target/*.jpi'
+                                    } else {
+                                        artifacts = '**/build/libs/*.hpi,**/build/libs/*.jpi'
+                                    }
+                                    archiveArtifacts artifacts: artifacts, fingerprint: true
                                 }
-                                archiveArtifacts artifacts: artifacts, fingerprint: true
                             }
                         }
+                    }            
+                } finally {
+                    deleteDir()
+
+                    if (env?.NODE_LABELS?.contains("docker")) {
```

Comment:
Or would suffice to just

```groovy
if (isUnix()) {
    sh 'docker system prune --force --all || :'
}
```

and let it just fail if Docker is not present.

Reply:
Every linux machine have docker installed

Reply:
does isUnix pass on ACI?
I've pushed @jglick 's suggested change but a little worried that it will just always fail on ACI, not that it matter's much as the command has `||` on it to make it never fail the build